crates/swc_ecma_transforms_typescript/src/transform.rs

```diff
@@ -649,92 +675,130 @@ impl Transform {
 
-            let value = Self::transform_ts_enum_member(
-                m,
-                &id.to_id(),
-                &default_init,
-                record,
-                self.unresolved_mark,
-            );
+        let member_list: Vec<_> = members
+            .into_iter()
+            .map(|m| {
+                let span = m.span;
+                let name = m.id.as_ref().clone();
 
-            default_init = value.inc();
-            {
                 let key = TsEnumRecordKey {
                     enum_id: id.to_id(),
                     member_name: name.clone(),
                 };
 
-                // Never store TsEnumRecordValue::Opaque value in the record.
-                let value = if value.is_const() {
-                    value.clone()
-                } else {
-                    TsEnumRecordValue::Void
-                };
-
-                record.insert(key, value);
-            }
-
-            let item = EnumMemberItem { span, name, value };
+                let value = self.enum_record.get(&key).unwrap().clone();
 
-            member_list.push(item);
-        }
-
-        let ts_enum_safe_remove = !self.verbatim_module_syntax && is_const && !is_export;
+                EnumMemberItem { span, name, value }
+            })
+            .filter(|m| !ts_enum_safe_remove || !m.is_const())
+            .collect();
 
-        if ts_enum_safe_remove && member_list.iter().all(EnumMemberItem::is_const) {
-            return Stmt::dummy();
+        if member_list.is_empty() && is_const {
+            return FoldedDecl::Empty;
         }
 
         let stmts = member_list
             .into_iter()
             .filter(|item| !ts_enum_safe_remove || !item.is_const())
-            .map(|item| item.build_assign(&id.to_id()))
-            .collect();
+            .map(|item| item.build_assign(&id.to_id()));
+
+        let namespace_export = self.namespace_id.is_some() && is_export;
+        let iife = !is_first || namespace_export;
+
+        let body = if !iife {
+            let return_stmt: Stmt = ReturnStmt {
+                arg: Some(id.clone().into()),
+                ..Default::default()
+            }
+            .into();
+
+            let stmts = stmts.chain(iter::once(return_stmt)).collect();
 
-        let expr = Self::wrap_enum_or_module_with_iife(
-            &self.namespace_id,
-            id,
             BlockStmt {
-                span: DUMMY_SP,
                 stmts,
                 ..Default::default()
-            },
-            is_export,
-        );
+            }
+        } else {
+            BlockStmt {
+                stmts: stmts.collect(),
+                ..Default::default()
+            }
+        };
 
-        ExprStmt {
-            span,
-            expr: expr.into(),
+        let var_kind = if is_export || id.ctxt == self.top_level_ctxt {
+            VarDeclKind::Var
+        } else {
+            VarDeclKind::Let
+        };
+
+        let init_arg = 'init_arg: {
```

Comment:
Does this work with our MSRV?

Reply:
This feature has been introduced for a long time, and I expect it to work on MSRV.
https://rust-lang.github.io/rfcs/2046-label-break-value.html